etcdctl/command/util.go

```diff
@@ -211,10 +211,18 @@ func mustNewClient(c *cli.Context) client.Client {
 				handleError(ExitServerError, err)
 			}
 
+			if cerr, ok := err.(*client.ClusterError); ok && cerr.IsBadConnection() {
+				handleError(ExitBadConnection, err)
+			}
+
 			// fail-back to try sync cluster with peer API. this is for making etcdctl work with etcd 0.4.x.
 			// TODO: remove this when we deprecate the support for etcd 0.4.
 			eps, serr := syncWithPeerAPI(c, ctx, hc.Endpoints())
 			if serr != nil {
+				// connections might go down when retrying for etcd 0.4
+				if cerr, ok := serr.(*client.ClusterError); ok && cerr.IsBadConnection() {
```

Comment:
probably we need to add another function like `isConnectionError(err error) bool`.